modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/ThriftClientFactory.java

```diff
@@ -96,4 +98,22 @@ public void destroyObject(PooledObject<SlotManagementService.Client> client) thr
         client.getObject().getInputProtocol().getTransport().close();
         super.destroyObject(client);
     }
-}
\ No newline at end of file
+
+    /**
+     * When setTestOnBorrow is set to true in thrift connection pool. This validation method is called on each
+     * connection object taken from the pool. This method do a bogus thrift call and validate the connection is
+     * usable for next intended task.
+     *
+     * @param client client
+     * @return connection's usability
+     */
+    @Override
+    public boolean validateObject(PooledObject<SlotManagementService.Client> client) {
+        try {
+            return client.getObject().healthCheck(Constants.HEALTH_CHECK);
+        } catch (TException e) {
+            log.error("exception occurs when validating the thrift connection " + e);
```

Comment:
It would be better to pass the exception rather than append. i.e,
 log.error("exception occurs when validating the thrift connection ", e);
WDYT?

Reply:
+1 changed